tests/integration/standard/test_query.py

```diff
@@ -551,9 +559,15 @@ def test_prepare_batch_statement(self):
 
         # This is going to query a host where the query
         # is not prepared
+        policy.set_host(1)
         batch_statement = BatchStatement(consistency_level=ConsistencyLevel.ONE)
         batch_statement.add(insert_statement, (1, 2))
         session.execute(batch_statement)
+
+        # To verify our test assumption that queries are getting re-prepared properly
+        self.assertEqual(1, self.mock_handler.get_message_count('debug', "Re-preparing"))
+
+        policy.set_host(2)
```

Comment:
What is the difference in this test, do we really need to use that custom policy? We could just use the `host` arg also?

Reply:
This particular line isn't necessary (I'll remove it), but the line on 562 is. The problem with this test is we need the batch statement to execute on a different host than the prepare statement. Unless there is a way to control the host that the prepare statement is going to, the policy is necessary for this test.

Reply:
oh I see now. That's the part I missed. Indeed we need it to ensure it will be applied when preparing. Thanks for the explanation!